catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/ConfigurationStore.java

```diff
@@ -83,4 +96,123 @@ public void setInMemory(boolean isInMemory) {
+  }
+
+  public void removeAnyTextWhiteListField(String whitelistField) {
+    this.anyTextFieldWhitelist.remove(whitelistField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void setAnyTextFieldBlacklist(List<String> fieldBlacklist) {
+    this.anyTextFieldBlacklist.clear();
+    if (fieldBlacklist != null) {
+      this.anyTextFieldBlacklist.addAll(fieldBlacklist);
+    }
+
+    filterAnyTextFieldCache();
+  }
+
+  public List<String> getAnyTextFieldBlacklist() {
+    return new ArrayList<>(anyTextFieldBlacklist);
+  }
+
+  public void addAnyTextBlacklistField(String blacklistField) {
+    this.anyTextFieldBlacklist.add(blacklistField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void removeAnyTextBlacklistField(String blacklistField) {
+    this.anyTextFieldBlacklist.remove(blacklistField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void addAnyTextField(String anyTextField) {
+    anyTextFieldsCache.add(anyTextField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void removeAnyTextField(String anyTextField) {
+    anyTextFieldsCache.remove(anyTextField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void clearAnyTextFieldCache() {
+    anyTextFieldsCache.clear();
+    filteredAnyTextFieldsCache.clear();
+  }
+
+  public Set<String> getAnyTextFieldsCache() {
+    return anyTextFieldsCache;
+  }
+
+  public Set<String> getFilteredAnyTextFields() {
+    return filteredAnyTextFieldsCache;
+  }
+
+  private void filterAnyTextFieldCache() {
+    Set<String> filteredList = new HashSet<>();
+
+    ConfigurationStore config = ConfigurationStore.getInstance();
+    List<String> anyTextFieldWhitelist = config.getAnyTextFieldWhitelist();
+    List<String> anyTextFieldBlacklist = config.getAnyTextFieldBlacklist();
+    if (!anyTextFieldWhitelist.isEmpty()) {
+      for (String whitelistField : anyTextFieldWhitelist) {
+        String whitelist;
+        if (!whitelistField.endsWith(SchemaFields.TEXT_SUFFIX)) {
+          whitelist = whitelistField + SchemaFields.TEXT_SUFFIX;
+        } else {
+          whitelist = whitelistField;
+        }
+        filteredList.addAll(
+            anyTextFieldsCache
+                .stream()
+                .filter(field -> field.matches(whitelist))
+                .collect(Collectors.toList()));
+      }
+    } else if (!anyTextFieldBlacklist.isEmpty()) {
+      filteredList.addAll(anyTextFieldsCache);
+      for (String blacklistField : anyTextFieldBlacklist) {
+        String blacklist;
+        if (!blacklistField.endsWith(SchemaFields.TEXT_SUFFIX)) {
+          blacklist = blacklistField + SchemaFields.TEXT_SUFFIX;
+        } else {
+          blacklist = blacklistField;
+        }
+        filteredList.removeAll(
+            anyTextFieldsCache
+                .stream()
+                .filter(field -> field.matches(blacklist))
+                .collect(Collectors.toList()));
+      }
+    } else {
+      filteredList.addAll(anyTextFieldsCache);
+    }
```

Comment:
I would suggest updating this to allow a blacklist and a whitelist at the same time.  A common strategy for this is have the whitelist take precedence.  Then admins could filter out entire namespaces while letting a few parts of the namespace through the filter.   

```suggestion
  private void filterAnyTextFieldCache() {
    Set<String> filteredList = new HashSet<>();

    ConfigurationStore config = ConfigurationStore.getInstance();
    List<String> anyTextFieldWhitelist = config.getAnyTextFieldWhitelist();
    List<String> anyTextFieldBlacklist = config.getAnyTextFieldBlacklist();
    
    if (!anyTextFieldBlacklist.isEmpty()) {
      filteredList.addAll(anyTextFieldsCache);
      for (String blacklistField : anyTextFieldBlacklist) {
        String blacklist;
        if (!blacklistField.endsWith(SchemaFields.TEXT_SUFFIX)) {
          blacklist = blacklistField + SchemaFields.TEXT_SUFFIX;
        } else {
          blacklist = blacklistField;
        }
        filteredList.removeAll(
            anyTextFieldsCache
                .stream()
                .filter(field -> field.matches(blacklist))
                .collect(Collectors.toList()));
      }
    } 
    if (!anyTextFieldWhitelist.isEmpty()) {
      for (String whitelistField : anyTextFieldWhitelist) {
        String whitelist;
        if (!whitelistField.endsWith(SchemaFields.TEXT_SUFFIX)) {
          whitelist = whitelistField + SchemaFields.TEXT_SUFFIX;
        } else {
          whitelist = whitelistField;
        }
        filteredList.addAll(
            anyTextFieldsCache
                .stream()
                .filter(field -> field.matches(whitelist))
                .collect(Collectors.toList()));
      }
    } 
    
    if (anyTextFieldWhitelist.isEmpty() && anyTextFieldBlacklist.isEmpty()) {
      filteredList.addAll(anyTextFieldsCache);
    }
```

Reply:
Your solution would allow non-whitelisted fields to be added to the anyText expansion since it starts with the full list, removes any blacklisted items, and then re-adds any whitelisted items.   That is really an implementation of a blacklist exception.  Wouldn't that be confusing to an admin where they provide a whitelist but still see non-whitelisted fields as part of the queries?

Reply:
Yeah, that's the intention in my solution.  You start with the full list, blacklist things, then add things there are exceptions for.  

I don't think it's confusing, it's a common pattern when both a blacklist and a whitelist are allowed.  

I would argue the current behavior is confusing, since we allow both a blacklist and a whitelist to be defined (with no warnings), and then ultimately only use the whitelist.

Reply:
If metatypes would allow me to force selecting only one of the lists at a time I would have done it for clarity.  I get your point.  Regardless of solution, it needs to identify to the user what the behavior is.  I will update it to allow both and add a comment to the whitelist description that will indicate it becomes a blacklist exception if both lists are provided instead of a pure whitelist.

Reply:
Sounds good.

I feel you on the metatypes, they feel like crude tools when it comes to the reality of configurations.

I might take a bit of time after we merge this into master to see if we can offer a special view in the new Admin UI for this configuration, might help ease the burden on admins.  I've been meaning to see if we can make a simple pattern for replacing the default configuration views.